src/docs/perf/rendering/ui-performance.md

```diff
@@ -405,46 +405,48 @@ Flutter's tools and debugging in Flutter:
-[stopwatch icon]: /docs/testing/ui-performance/images/observatory-icon.png
+[Flutter's build modes]: /docs/testing/build-modes
+[GitHub wiki]: {{site.github}}/flutter/flutter/wiki/
+[Inspector view]: /docs/development/tools/devtools/inspector
+[Integration testing]: /docs/testing#integration-tests
 [issues or feature requests]: {{site.github}}/dart-lang/sdk/issues?q=is%3Aopen+is%3Aissue+label%3Aarea-observatory
+[line-chart icon]: /docs/testing/ui-performance/images/observatory-timeline-icon.png
+[MainThread]: {{site.android-dev}}/reference/android/support/annotation/MainThread
+[Performance overlay]: /docs/testing/code-debugging#performance-overlay
+[PerformanceOverlay]: {{site.api}}/flutter/widgets/PerformanceOverlay-class.html
+[PerformanceOverlayLayer.checkerboardOffscreenLayers]: {{site.api}}/flutter/rendering/PerformanceOverlayLayer/checkerboardOffscreenLayers.html
+[PerformanceOverlayLayer.checkerboardRasterCacheImages]: {{site.api}}/flutter/rendering/PerformanceOverlayLayer/checkerboardRasterCacheImages.html
+[profile mode]: /docs/testing/build-modes#profile
+[programmatically]: /docs/testing/code-debugging#debugging-animations
+[rendering library]: {{site.api}}/flutter/rendering/rendering-library.html
+[RepaintBoundary]: {{site.api}}/flutter/widgets/RepaintBoundary-class.html
 [`saveLayer`]: {{site.api}}/flutter/dart-ui/Canvas/saveLayer.html
-[timeDilation]: {{site.api}}/flutter/scheduler/timeDilation.html
 [Show performance data]: /docs/development/tools/android-studio#show-performance-data
-[Android Studio/IntelliJ]: /docs/development/tools/android-studio
-[rendering library]: {{site.api}}/flutter/rendering/rendering-library.html
-[Testing Flutter apps]: /docs/testing
-[Why Flutter Uses Dart]: https://hackernoon.com/why-flutter-uses-dart-dd635a054ebf
-[devtools]: /docs/development/tools/devtools
-[Inspector view]: /docs/development/tools/devtools/inspector
-[Timeline view]: /docs/development/tools/devtools/timeline
-[programmatically]: /docs/testing/code-debugging#debugging-animations
-[Performance overlay]: /docs/testing/code-debugging#performance-overlay
-[Debugging Flutter apps programmatically]: /docs/testing/code-debugging
-[launch DevTools]: /docs/development/tools/devtools
+[stopwatch icon]: /docs/testing/ui-performance/images/observatory-icon.png
 [The Layer Cake]: https://medium.com/flutter-community/the-layer-cake-widgets-elements-renderobjects-7644c3142401
-[GitHub wiki]: {{site.github}}/flutter/flutter/wiki/
 [The Framework architecture]: {{site.github}}/flutter/flutter/wiki/The-Framework-architecture
+[timeDilation]: {{site.api}}/flutter/scheduler/timeDilation.html
+[Tracing Dart code]: /docs/testing/debugging#tracing-dart-code
+[Testing Flutter apps]: /docs/testing
+[Timeline view]: /docs/development/tools/devtools/timeline
+[UIKit]: https://developer.apple.com/documentation/uikit
+[Why Flutter Uses Dart]: https://hackernoon.com/why-flutter-uses-dart-dd635a054ebf
```

Comment:
Delete? Or if it's still used somewhere, change to something more obviously the article rather than the video...

Reply:
It's a good article (Wm wrote it), but it does have one broken image. I'm conflicted, because it goes into more depth than the video...

Reply:
I'm going to leave both for now.